src/json-data.js

```diff
@@ -1,51 +1,60 @@
-const fs = require('fs-extra');
-const path = require('path');
-const reserved = require('reserved-words');
-const shared = require('./shared');
-const glob = require('glob');
+const cheerio = require('cheerio');
+const jsonDataUtil = require('./utils/json-data');
+const shared = require('./utils/shared');
 
 const preload = (content, resourcePath) => {
   if (resourcePath.match(/\.html$/)) {
-    return content.toString().replace(/stache\.jsonData\./g, 'stache.jsonData?.');
+    return editHTMLContent(content);
   }
 
-  if (!resourcePath.match(/app-extras\.module\.ts$/)) {
-    return content;
+  if (resourcePath.match(/app-extras\.module\.ts$/)) {
+    return addGlobalDataToAppExtrasModule(content, resourcePath);
   }
 
-  const root = shared.resolveAssetsPath('data');
-  const filePaths = glob.sync(path.join(root, '*.json'));
+  return content;
+};
+
+const editHTMLContent = (content) => {
+  const $ = cheerio.load(content, shared.cheerioConfig);
+  const stacheTags = $('stache');
 
-  if (!filePaths.length) {
-    return content;
+  if (stacheTags.length > 0) {
+    content = parseStacheAttributeBindings(stacheTags, $);
   }
 
-  const dataObject = filePaths.reduce((acc, filePath) => {
-    const fileName = path.basename(filePath);
-    const propertyName = convertFileNameToObjectPropertyName(fileName);
-
-    if (!isPropertyNameValid(propertyName)) {
-      console.error(
-        new shared.StachePluginError(
-          `A valid Object property could not be determined from file ${fileName}! The property key '${propertyName}' cannot be used. Please choose another file name.`
-        )
-      );
-      return acc;
-    }
+  content = jsonDataUtil.parseAllBuildTimeBindings($.html().toString());
```

Comment:
For some reason, adding `@buildtime:` to Page Anchor Components didn't seem to work. 

Also, since this file is getting fairly large, I'm curious if pulling the "buildtime" specific parsing into a separate plugin would be beneficial (`json-data-buildtime` or something), to keep concerns separated, and to ease debugging.